addons/website_sale/controllers/main.py

```diff
@@ -997,3 +997,29 @@ def country_infos(self, country, mode, **kw):
             states=[(st.id, st.name, st.code) for st in country.get_website_sale_states(mode=mode)],
             phone_code=country.phone_code
         )
+
+    @http.route("/shop/cart/<token>", type='http', auth="public", website=True)
+    def access_abandoned_cart(self, token=None, **post):
+        """
+        Token will be pass from revival email template and it will redirect to the selected cart.
+        """
+        order = request.env['sale.order'].sudo().search([('access_token', '=', token)], limit=1)
+        if not order:
+            return request.render('website.404')
+        if order.state != 'draft': # check that cart has been procced or not.
+            response = self.cart()
+            response.qcontext.update({'proceed': 1})
+            return response
+        if post.get('new'):  # redirect to previous cart
```

Comment:
Well I would get it out of the **post and make it a real argument, as it is quite an important method parameter. I think it could be better named, because grepping 'new' in Odoo when you want to find where it is used and/or defined will give you a lot of results.